python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+        ----------
+        time: mandatory, float
+            MJD at which band fluxes are evaluated
+        bandpassDict: mandatory, `lsst.sims.photUtils.BandpassDict` instance
+        observedBandPassInd : optional, list of integers, defaults to None
+            integer correspdonding to index of the bandpasses used in the
+            observation in the ordered dict bandpassDict
+        Returns
+        -------
+        `~numpy.ndarray` of length =len(observedBandPassInd)
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+        SEDfromSNcosmo = self.SNObjectSED(time=time,
+                                          bandpass=bandpassDict['u'])
+        wavelen_step = np.diff(SEDfromSNcosmo.wavelen)[0]
+        SEDfromSNcosmo.flambdaTofnu()
+        f = SEDfromSNcosmo.manyFluxCalc(bandpassDict.phiArray,
+                                        wavelen_step=wavelen_step,
+                                        observedBandpassInd=observedBandPassInd)
+        return f / 3631.
+
+    def catsimManyBandMags(self, time, bandpassDict,
+                           observedBandPassInd=None):
+        """
+        return the flux in the bandpass in units of the flux
+        the AB magnitude reference spectrum would have in the
+        same band.
+
+        Parameters
+        ----------
+        time: mandatory, float
+            MJD at which band fluxes are evaluated
+        bandpassDict: mandatory, `lsst.sims.photUtils.BandpassDict` instance
+        observedBandPassInd : optional, list of integers, defaults to None
+            integer correspdonding to index of the bandpasses used in the
+            observation in the ordered dict bandpassDict
+        Returns
+        -------
+        `~numpy.ndarray` of length =len(observedBandPassInd)
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+        f = self.catsimManyBandFluxes(time,
+                                      bandpassDict,
+                                      observedBandPassInd)
+
+        return -2.5 * np.log10(f)
+
+    def catsimBandMags(self, bandpassobject, time):
+        """
+        return the magnitude in the bandpass in the AB magnitude system
+
+        Parameters
+        ----------
+        bandpassobject : mandatory,`sims.photUtils.BandPass` instances
+            LSST Catsim bandpass instance for the bandpass
+        time : mandatory, float
+              MJD at which this is evaluated
+        Returns
+        -------
+            float
+        Examples
+        --------
+        """
+        fluxRatio = self.catsimBandFluxes(bandpassobject=bandpassobject,
+                                          time=time)
+        return -2.5 * np.log10(fluxRatio)
+
+    def catsimADU(self, time, bandpassDict,
```

Comment:
I think you should keep the same convention you established for magnitudes and fluxes:

catsimBandADU calculates the ADU in one bandpass

catsimManyBandADU calculates the ADU in all the bandpasses of a BandpassDict